plugins/events-backend-module-kafka/src/publisher/LoggerServiceAdapter.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { LogEntry, logLevel } from 'kafkajs';
+
+export const loggerServiceAdapter =
+  (loggerService: LoggerService) => (_level: logLevel) => {
+    return (entry: LogEntry) => {
+      const { namespace, level, log } = entry;
+      const { message, ...extra } = log;
+
+      const logMethods: Record<
```

Comment:
this could have been placed inside the outermost function to not have to reconstruct it over and over, but that can be done as a follow-up and is no biggie